src/pytorch_lightning/utilities/deepspeed_model_summary.py

```diff
@@ -35,12 +36,24 @@ class DeepSpeedLayerSummary(LayerSummary):
     @property
     def num_parameters(self) -> int:
         """Returns the number of parameters in this module."""
-        return sum(deepspeed_param_size(p) if not _is_lazy_weight_tensor(p) else 0 for p in self._module.parameters())
+        if isinstance(self._module, torch.nn.Module):
+            return sum(
+                deepspeed_param_size(p) if not _is_lazy_weight_tensor(p) else 0 for p in self._module.parameters()
+            )
+        elif isinstance(self._module, torch.nn.Parameter):
```

Comment:
I would handle this separately. The docstring says it handles the numver of parameters in **this module** so the module itself shouldn't be a nn.Parameter .

Also the class is named `LayerSummary`, to me that looks like directly registered parameters should be handled externally as well.

Reply:
The class LayerSummary was originally the class that summarizes a single layer whose type was nn.Module.
However, not to ignore nn.Parameter I changed the type of layer to be nn.Module or nn.Parameter.
Because it can be nn.Module and nn.Parameter, I should write as above.

I think I should make another class that summarizes nn.Parameter and it will be class ParameterSummary.